File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */


// class TreeNode {
//     constructor(val = 0, left = null, right = null) {
//         this.val = val;
//         this.left = left;
//         this.right = right;
//     }
// }

class Solution {
    /**
     * @param {TreeNode} root
     * @return {TreeNode}
     */
    invertTree(root) {
        if (root === null) {
            return root
        }
        let originalLeft = root.left;
        let originalRight = root.right;

        if (root.left) {
            this.invertTree(root.left)
        }
        root.left = originalRight

        if (root.right) {
            this.invertTree(root.right)
        }
        root.right = originalLeft

        return root
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        self.res = 0

        def dfs(root, maximum):
            if not root:
                return 
            if root.val >= maximum:
                self.res += 1

            left = dfs(root.left, max(maximum, root.val))
            right = dfs(root.right, max(maximum, root.val))

        dfs(root, root.val)

        return self.res